draft-ietf-wish-whip.md

```diff
@@ -242,9 +244,9 @@ Trickle ICE and ICE restart support are RECOMMENDED for both WHIP sessions and c
 
 The WHIP client MAY perform trickle ICE or ICE restarts by sending an HTTP PATCH request as per {{!RFC5789}} to the WHIP session URL, with a body containing an SDP fragment with media type "application/trickle-ice-sdpfrag" as specified in {{!RFC8840}} carrying the relevant ICE information.
 
-If the HTTP PATCH to the WHIP session has a content type different than "application/trickle-ice-sdpfrag", the WHIP session MUST reject the HTTP PATCH request with a "415 Unsupported Media Type" error response. If the SDP fragment is malformed, the WHIP session MUST reject the HTTP PATCH with a "400 Bad Request" error response.
+If the HTTP PATCH to the WHIP session has a content type different than "application/trickle-ice-sdpfrag" or the SDP fragment is malformed, the WHIP session MUST reject the HTTP PATCH request with an 4XX error response.
 
-If the WHIP session supports either Trickle ICE or ICE restarts, but not both, it MUST return a "422 Unprocessable Content" response for the HTTP PATCH requests that are not supported as per {{Section 15.5.21 of !RFC9110}}. 
+If the WHIP session supports either Trickle ICE or ICE restarts, but not both, it MUST return an 5XX error response for the HTTP PATCH requests that are not supported as per {{Section 15.5.21 of !RFC9110}}. 
```

Comment:
501 is only used when the origin server does not support a method for ANY resource on the origin server.  I don't believe this is the case here.
I think using 422 is a reasonable choice here to convey that request cannot be handled (for some reason).  The BCP56 guidance is recommending that the specification doesn't equate 422 to "Trickle ICE or ICE restart is not supported".  The specification should not restrict 422 from being used for other application level errors that fall within the semantics of "unprocessable entity".   If it is believed that the client needs to know their request failed because the server doesn't support their specific kind of request, e.g. so they can try the alternative automatically, then it may be wise to communicate the specifics via an HTTP problem response. This would allow for the use of 422 to describe other kinds of application errors in the future.